src/patternfly/layouts/Flex/flex.scss

```diff
@@ -118,274 +15,298 @@ $pf-flex-spacer-map: (
   --pf-l-flex--spacer: var(--pf-l-flex--spacer-base);
   --pf-l-flex--flex--spacer: var(--pf-global--spacer--lg);
 
-  // Flex spacer values
-  --pf-l-flex--spacer--none: 0;
-  --pf-l-flex--spacer--xs: var(--pf-global--spacer--xs);
-  --pf-l-flex--spacer--sm: var(--pf-global--spacer--sm);
-  --pf-l-flex--spacer--md: var(--pf-global--spacer--md);
-  --pf-l-flex--spacer--lg: var(--pf-global--spacer--lg);
-  --pf-l-flex--spacer--xl: var(--pf-global--spacer--xl);
-  --pf-l-flex--spacer--2xl: var(--pf-global--spacer--2xl);
-  --pf-l-flex--spacer--3xl: var(--pf-global--spacer--3xl);
+  // Emit spacer css variables that map to requested spacer values
+  @include pf-emit-properties($pf-l-flex--variable-map);
 
   display: var(--pf-l-flex--Display);
   flex-wrap: var(--pf-l-flex--FlexWrap);
   align-items: var(--pf-l-flex--AlignItems);
+}
 
-  // Establish default item spacing
-  > * {
-    margin-right: var(--pf-l-flex--spacer);
+/* stylelint-disable no-duplicate-selectors, max-nesting-depth */
 
-    &:last-child:not([class*="pf-m-spacer"]) {
-      --pf-l-flex--spacer: 0;
-    }
-  }
+// Flex
+.pf-l-flex {
+  --pf-l-flex--spacer: var(--pf-l-flex--spacer-base);
 
-  // Reset item spacing in nested flex layouts
-  & & {
-    --pf-l-flex--spacer: var(--pf-l-flex--spacer-base);
+  &:last-child {
+    --pf-l-flex--spacer: 0;
+  }
+}
 
-    > * {
-      --pf-l-flex--spacer: var(--pf-l-flex--spacer-base);
-    }
+// Item
+.pf-l-flex > *,
+.pf-l-flex__item {
+  // reset spacer
+  --pf-l-flex--spacer: var(--pf-l-flex--spacer-base);
 
-    > :last-child:not([class*="pf-m-spacer"]),
-    &:last-child:not([class*="pf-m-spacer"]) {
-      --pf-l-flex--spacer: 0;
-    }
+  &:last-child {
+    --pf-l-flex--spacer: 0;
   }
+}
 
-  @include pf-l-flex--space-items-builder;
-  @include pf-l-flex--spacer-builder;
-
-  // Breakpoints with responsive modifiers
-  $breakpoint-list: (
-    (null),
-    ($pf-global--breakpoint--sm, "-on-sm"),
-    ($pf-global--breakpoint--md,"-on-md"),
-    ($pf-global--breakpoint--lg, "-on-lg"),
-    ($pf-global--breakpoint--xl, "-on-xl"),
-    ($pf-global--breakpoint--2xl, "-on-2xl")
-  );
-
-  // List of modifiers
-  @mixin flex-modifiers($mod: null) {
-    // display
-    &.pf-m-flex#{$mod} {
-      display: var(--pf-l-flex--Display);
-    }
+// Use global selector to
+.pf-l-flex > *,
+.pf-l-flex__item {
+  margin-right: var(--pf-l-flex--spacer);
+}
 
-    &.pf-m-inline-flex#{$mod} {
-      --pf-l-flex--Display: inline-flex;
-    }
+.pf-l-flex {
+  @each $breakpoint, $breakpoint-value in $pf-l-flex--breakpoint-map {
+    $breakpoint-name: if($breakpoint != "base", -on-#{$breakpoint}, "");
+
+    @include pf-apply-breakpoint($breakpoint, $pf-l-flex--breakpoint-map) {
+      // display
+      &.pf-m-flex#{$breakpoint-name},
+      .pf-m-flex#{$breakpoint-name} {
```

Comment:
I don't think we want this last selector, that will target `.pf-l-flex .pf-m-flex` and we could have a `pf-m-flex` for some other component.